### PR TITLE
Add git-cola

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/git-cola/git-cola/issues/1224
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. (https://github.com/git-cola/git-cola/pull/1225)

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.git-cola.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+app-id: com.github.git-cola
+runtime: org.kde.Platform
+runtime-version: '5.15'
```
Comment (hfiguiere) [inline com.github.git-cola.yml:3]:
```suggestion
runtime-version: '5.15-21.08'
```


5.15 is EOL

File: com.github.git-cola.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --filesystem=home
+
+modules:
+  - name: pyside2
+    buildsystem: simple
```
Comment (hfiguiere) [inline com.github.git-cola.yml:17]: you should be able to use `buildsystem: cmake-ninja` and have the first line in a `shell` source. that was always run at unpack time.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.git-cola.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+app-id: com.github.git-cola
```
Comment (hfiguiere) [inline com.github.git-cola.yml:1]: also the app-id should be more like `com.github.git_cola.git-cola`

File: com.github.git-cola.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+runtime: org.kde.Platform
+runtime-version: '5.15'
+sdk: org.kde.Sdk
+base: io.qt.qtwebkit.BaseApp
+base-version: "5.15"
+command: git-cola
```
Comment (hfiguiere) [inline com.github.git-cola.yml:7]: you also likely need a `rename-desktop-file` and `rename-appdata-file`, and change the app-dir in the appdata file from upstream.

Note sure about the icons, but likely as well.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build com.github.git-cola

File: com.github.git-cola.yml (hunk 1 above)
Comment (Mailaender) [reply com.github.git-cola.yml:3]: -21.08 is missing Python.

Comment (hfiguiere) [reply com.github.git-cola.yml:3]: not it's not. It's python 3.9

5.15 is EOL anyway.

File: com.github.git-cola.yml (hunk 4 above)
Comment (hfiguiere) [reply com.github.git-cola.yml:7]: yeah it needs a `rename-icon: git-cola` as well.

However the appdata doesn't seem to be installed by the build

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.git-cola.yml (hunk 3 above)
Comment (Mailaender) [reply com.github.git-cola.yml:1]: That sounds horrible.

**Comment by Mailaender** [comment]
bot, build com.github.git-cola

File: com.github.git-cola.yml (hunk 1 above)
Comment (Mailaender) [reply com.github.git-cola.yml:3]: I misinterpreted the error. It was missing LLVM, but there is an extension for that.

**Comment by Mailaender** [comment]
bot, build com.github.git-cola

File: com.github.git-cola.yml (hunk 3 above)
Comment (hfiguiere) [reply com.github.git-cola.yml:1]: It is what it is https://docs.flatpak.org/en/latest/conventions.html#application-ids

**Comment by Mailaender** [comment]
bot, build com.github.git-cola

**Comment by hfiguiere** [comment]
I don't think `rename-icon` was the problem. From what I can see the install rule *symlink* and that's what I suspect is causing rename-icon to fail.

**Comment by hfiguiere** [comment]
Also another failure that will pop:
```
repo/files/share/appdata/com.github.git-cola.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <release> required
Validation of files failed
```

**Comment by hfiguiere** [comment]
So to fix the icon problem, the follow patch works:
```diff
diff --git a/com.github.git-cola.yml b/com.github.git-cola.yml
index 7a47afa..74815d4 100644
--- a/com.github.git-cola.yml
+++ b/com.github.git-cola.yml
@@ -9,6 +9,7 @@ sdk-extensions:
 command: git-cola
 rename-desktop-file: git-cola.desktop
 rename-appdata-file: git-cola.appdata.xml
+rename-icon: git-cola
 finish-args:
   - --share=ipc
   - --share=network
@@ -43,6 +44,8 @@ modules:
     build-commands:
       - make install prefix=${FLATPAK_DEST}
       - cp -r share ${FLATPAK_DEST}
+      - rm ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/git-cola.svg
+      - cp ${FLATPAK_DEST}/share/git-cola/icons/git-cola.svg  ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/
       - install -Dm0755 $(which git) ${FLATPAK_DEST}/bin/
     sources:
       - type: git
```

As I mentionned earlier the `rename-icon` isn't the problem. The symlink is.

**Comment by Mailaender** [comment]
bot, build com.github.git-cola

**Comment by hfiguiere** [comment]
> bot, build com.github.git-cola

This is why the build fail. The manifest name sould match the app-id.

**Comment by Mailaender** [comment]
bot, build com.github.git-cola

**Comment by Mailaender** [comment]
bot, build com.github.git_cola.git-cola

**Comment by hfiguiere** [comment]
bot, build com.github.git_cola.git-cola

**Pull request closed without merging**